hpx/runtime/serialization/detail/polymorphic_nonintrusive_factory.hpp

```diff
@@ -189,15 +193,47 @@ namespace hpx { namespace serialization { namespace detail
 #define HPX_SERIALIZATION_REGISTER_CLASS_NAME(Class, Name)                    \
     namespace hpx { namespace serialization { namespace detail {              \
         template <> HPX_ALWAYS_EXPORT                                         \
-        char const* get_serialization_name<Class>()                           \
+        struct get_serialization_name<Class>                                  \
         {                                                                     \
-            return Name;                                                      \
-        }                                                                     \
+            char const* operator()()                                          \
+            {                                                                 \
+                return Name;                                                  \
+            }                                                                 \
+        };                                                                    \
     }}}                                                                       \
     template hpx::serialization::detail::register_class<Class>                \
         hpx::serialization::detail::register_class<Class>::instance;          \
 /**/
+#define HPX_SERIALIZATION_REGISTER_CLASS_NAME_TEMPLATE(Parameters, Template, Name) \
+    namespace hpx { namespace serialization { namespace detail {              \
+        Parameters                                                            \
```

Comment:
Thanks Andreas. AFAICS, those others you moved were not misplaced before as they had been applied to functions, not type definitions.